### ISSUE TITLE
Game doesn't use Mac OS X fullscreen

### ISSUE THREAD

**Issue by MarioSMB** [Unvanquished/Unvanquished#790]
Unsure if this is intentional, but the game doesn't switch to a new space in full screen mode (most apps create a desktop space you can switch to with CTRL+arrow keys), but instead stays in the current workspace, making it a little harder to switch workspaces.

Disclaimer:
I was unable to find the details online of modern Mac's fullscreen desktop switching, sorry if this isn't enough information...

**Comment by DolceTriade** [comment]
Do most games also do this? ie, steam games?

**Comment by MarioSMB** [comment]
All I can really test currently is Xonotic, and it does switches desktops properly.

**Comment by MarioSMB** [comment]
Alright, I tested a few more games. Most of them do indeed switch spaces (seems to be random though, guess developers need to know how to enable it).

**Comment by DolceTriade** [comment]
I see. Then I will leave this as an open issue. We'll try to get around to it for when we start polishing the game.

**Comment by Amanieu** [comment]
Could this be due to the difference between SDL_WINDOW_FULLSCREEN and
SDL_WINDOW_FULLSCREEN_DESKTOP?

https://wiki.libsdl.org/SDL_SetWindowFullscreen

We always use SDL_WINDOW_FULLSCREEN but darkplaces can choose one or the
other using vid_desktopfullscreen.

**Comment by DolceTriade** [comment]
Yes, it seems to be. However, using SDL_WINDOW_FULLSCREEN_DESKTOP seems to have weird effects if your game resolution is different from your desktop resolution. And sometimes, the top menu bar seems to reappear even if r_fullscreen is 1. Is the abillity to change desktops using the trackpad a huge benefit?

**Comment by Kangz** [comment]
I think that if we can, fullscreen desktop is better (provided we match the monitor's resolution)

**Comment by MarioSMB** [comment]
Ideally, this would be optional, just like in DarkPlaces.  
The effects of fullscreen desktop are very much desired in setups like mine, where the game's idea of full screen is 60hz, while my desktop is at 120hz.

**Comment by slipher** [comment]
With SDL 3, a desktop space is automatically created when in fullscreen mode.
